### PR TITLE
Oksana

### PR DESCRIPTION
**Pull request by OxiBo**
update a new artist whose profile was initially rejected

### CODE REVIEW

File: applications/directory/src/components/CreateProfileForm.js [hunk 1]
```diff
@@ -636,7 +647,10 @@ const CreateProfileForm = ({
                     />
                   ))
                 }
-                onChange={(event, value) => setFieldValue('categories', value)}
+                onChange={(event, value, reason) => {
+                  console.log(reason);
```
Comment (chaddjohnson) [inline applications/directory/src/components/CreateProfileForm.js]: Please remove `console.log()` call.

File: services/directory-api/src/routes/updateNewArtist.js [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    await new Promise((resolve) => setTimeout(resolve, 25));
+    return 'Lambda is warm!';
+  }
+
+  try {
+    const { editProfileToken } = event.pathParameters;
+    const data = JSON.parse(event.body);
+
+    const updatedArtist = await httpClient.patch(
+      `/artists/edit/edit-profile-token/${editProfileToken}`,
```
Comment (chaddjohnson) [inline services/directory-api/src/routes/updateNewArtist.js]: This path should be `/artists/edit-profile-token/${editProfileToken}` (no `/edit`).

File: services/artists-api/serverless.yml [hunk 3]
```diff
@@ -118,6 +118,14 @@ functions:
           path: /reviews/token/{reviewToken}
           method: patch
           authorizer: aws_iam
+  updateNewArtist:
+    role: ${env:SERVICES_CONSUMER_ROLE_ARN, ssm:/artistdirectory/${self:provider.stage}/roles/services-consumer-arn}
+    handler: src/routes/updateNewArtist.handler
+    events:
+      - http:
+          path: /artists/edit/edit-profile-token/{editProfileToken}
```
Comment (chaddjohnson) [inline services/artists-api/serverless.yml]: This path should be `/artists/edit-profile-token/${editProfileToken}` (not `/edit`).

File: applications/directory/src/components/CreateProfileForm.js [hunk 4]
```diff
@@ -335,6 +341,8 @@ const CreateProfileForm = ({
         if (Object.keys(artist).length !== 0) {
           console.log(data);
           // TODO - send data to backend and update the artist
+          const art = await updateNewArtist(data, artist.editProfileToken);
+          console.log(art);
```
Comment (chaddjohnson) [inline applications/directory/src/components/CreateProfileForm.js]: Need to remove `console.log()`.

File: applications/directory/src/components/CreateProfileForm.js [hunk 5]
```diff
@@ -335,6 +341,8 @@ const CreateProfileForm = ({
         if (Object.keys(artist).length !== 0) {
           console.log(data);
           // TODO - send data to backend and update the artist
+          const art = await updateNewArtist(data, artist.editProfileToken);
```
Comment (chaddjohnson) [inline applications/directory/src/components/CreateProfileForm.js]: Once `console.log()` is removed, no need for `const art` here.

File: applications/directory/src/components/CreateProfileForm.js [hunk 6]
```diff
@@ -335,6 +341,8 @@ const CreateProfileForm = ({
         if (Object.keys(artist).length !== 0) {
           console.log(data);
```
Comment (chaddjohnson) [inline applications/directory/src/components/CreateProfileForm.js]: Also let's remove this `console.log()`.

**Review by chaddjohnson — CHANGES_REQUESTED** [review]

**Comment by chaddjohnson** [comment]
@OxiBo Please see refactored changes here: https://github.com/neatowebsolutions/artistdirectory/commit/3176fd3a1742106357653ab5a8b1a515851e0af6

File: services/artists-api/serverless.yml (hunk 3 above)
Comment (OxiBo) [reply services/artists-api/serverless.yml]: that is correct. Forgot to change it. It was part of debugging process

File: services/directory-api/src/routes/updateNewArtist.js (hunk 2 above)
Comment (OxiBo) [reply services/directory-api/src/routes/updateNewArtist.js]: that is correct. Forgot to change it. It was part of debugging process

File: applications/directory/src/components/CreateProfileForm.js (hunk 5 above)
Comment (OxiBo) [reply applications/directory/src/components/CreateProfileForm.js]: Also was part of debugging. Will be deleted

File: applications/directory/src/components/CreateProfileForm.js (hunk 6 above)
Comment (OxiBo) [reply applications/directory/src/components/CreateProfileForm.js]: ok

File: applications/directory/src/components/CreateProfileForm.js (hunk 1 above)
Comment (OxiBo) [reply applications/directory/src/components/CreateProfileForm.js]: done

**Pull request merged**